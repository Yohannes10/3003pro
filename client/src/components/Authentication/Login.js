import React, { useState } from "react";
import { Link, useNavigate } from "react-router-dom";
import avatar from "./assets/profile.png";
import toast, { Toaster } from "react-hot-toast";
import { useFormik } from "formik";
import styles from "../styles/Username.module.css";


const Login = ({ handleLogin }) => {
  // State variables to manage the username and password input values
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const formik = useFormik({});


  // Function to handle the form submission
  const handleSubmit = (e) => {
    // Prevent the default form submission behavior
    e.preventDefault();
    // Call the 'handleLogin' function provided by the parent component, passing the username and password as arguments
    handleLogin(username, password);
  };


  return (
    <div className="container mx-auto">
      <Toaster position="top-center" reverseOrder={false} />

      <div className="flex justify-center items-center h-screen">
        <div className={styles.glass} style={{ width: '45%', height: '90%' }}>
          <div className="title flex flex-col items-center" >
            <h4 className="text-5xl font-bold">
              Hello
            </h4>
            <span className="py-4 text-xl w-2/3 text-center text-gray-500">
              Explore More by connecting with us.
            </span>
          </div>

          <form className="py-1" onSubmit={handleSubmit}>
            <div className="profile flex justify-center py-4">
              <img src={avatar} className={styles.profile_img} alt="avatar" />
            </div>

            <div className="textbox flex flex-col items-center gap-6">
              <input
                className={styles.textbox}
                type="text"
                placeholder="Username"
                onChange={(e) => setUsername(e.target.value)}
              />
              <input
                className={styles.textbox}
                type="Password"
                placeholder="Password"
                onChange={(e) => setPassword(e.target.value)}
              />
              <button className={styles.btn} type="submit">
                Sign In
              </button>
            </div>

            <div className="text-center py-4">
              <span className="text-gray-500">
                Forgot Password?{" "}
                <Link className="text-red-500" to="/">
                  Recover Now
                </Link>
              </span>
            </div>
            <div className="text-center py-4">
              <span className='text-gray-500'>Not a Member <Link className='text-red-500' to="/register">Register Now</Link></span>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
}

export default Login;
