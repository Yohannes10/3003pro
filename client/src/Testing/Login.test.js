import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Login from "../components/Authentication/Login";
import { MemoryRouter } from 'react-router-dom';

test('renders Login component', () => {
  const { asFragment } = render(<MemoryRouter><Login handleLogin={() => {}} /></MemoryRouter>);

  expect(screen.getByPlaceholderText('Username')).toBeInTheDocument();
  expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
  expect(screen.getByText('Sign In')).toBeInTheDocument();

  // Perform snapshot testing
  expect(asFragment()).toMatchSnapshot();
});

test('handles form submission', () => {
  const mockHandleLogin = jest.fn();

  render(<MemoryRouter><Login handleLogin={mockHandleLogin} /></MemoryRouter>);

  const usernameInput = screen.getByPlaceholderText('Username');
  const passwordInput = screen.getByPlaceholderText('Password');

  fireEvent.change(usernameInput, { target: { value: 'testuser' } });
  fireEvent.change(passwordInput, { target: { value: 'testpassword' } });

  const signInButton = screen.getByText('Sign In');
  fireEvent.click(signInButton);

  expect(mockHandleLogin).toHaveBeenCalledWith('testuser', 'testpassword');
});
