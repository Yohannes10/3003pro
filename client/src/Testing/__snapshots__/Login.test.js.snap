// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login component 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="flex justify-center items-center h-screen"
    >
      <div
        class="glass"
        style="width: 45%; height: 90%;"
      >
        <div
          class="title flex flex-col items-center"
        >
          <h4
            class="text-5xl font-bold"
          >
            Hello
          </h4>
          <span
            class="py-4 text-xl w-2/3 text-center text-gray-500"
          >
            Explore More by connecting with us.
          </span>
        </div>
        <form
          class="py-1"
        >
          <div
            class="profile flex justify-center py-4"
          >
            <img
              alt="avatar"
              class="profile_img"
              src="profile.png"
            />
          </div>
          <div
            class="textbox flex flex-col items-center gap-6"
          >
            <input
              class="textbox"
              placeholder="Username"
              type="text"
            />
            <input
              class="textbox"
              placeholder="Password"
              type="text"
            />
            <button
              class="btn"
              type="submit"
            >
              Sign In
            </button>
          </div>
          <div
            class="text-center py-4"
          >
            <span
              class="text-gray-500"
            >
              Forgot Password? 
              <a
                class="text-red-500"
                href="/"
              >
                Recover Now
              </a>
            </span>
          </div>
          <div
            class="text-center py-4"
          >
            <span
              class="text-gray-500"
            >
              Not a Member 
              <a
                class="text-red-500"
                href="/register"
              >
                Register Now
              </a>
            </span>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
